gpu/octree/src/cuda/knn_search.cu

```diff
@@ -282,6 +285,7 @@ void pcl::device::OctreeImpl::nearestKSearchBatch(const Queries& queries, int /*
     
     batch.output = results.data;
     batch.sizes  = results.sizes;
+    batch.sqr_distances = sqr_distances;
```

Comment:
This looks very fishy because  batch.sqr_distances is a float pointer and `BatchResultSqrDists` is a `DeviceArray<float>` which is a full blow object.

I'm gonna assuming you're casting the objects memory to this pointer and that's why this works.